### PR TITLE
Add switch forces button

### LINKED ISSUE #862: Switch forces
The analysts know which participants are in a serial, but they don't currently know which ones will be "playing" as Red.

They have added them to the serials, but know they'll have to switch sides for quite a lot of them.

It would be good if Serial Participants could have a `Switch Forces` button.  This will switch them from red->blue (or vice versa) and update the Serial UI form.

### PR DESCRIPTION
**Pull request by robintw**
## 🧰 Issue
Fixes #862 

## 🚀 Overview: 
Add a button to switch the force of a participant - ie. move it from Red to Blue and vice versa.

## 🔗 Link to preview (or screenshot, if relevant)
image

## 🤔 Reason: 
Make it easier to switch the force of a participant without deleting and re-adding.

## 🔨Work carried out:

- [x] Add button and handler
- [x] Tests pass

## Confirmations

- [x] I have chosen reviewers for my PR.
- [x] I have chosen an appropriate label for the PR, adding `interactive_review` if reviewers will need to see UI
- [x] I have extended/updated the documentation in `\docs` folder
- [x] Any database content changes (Create, Edit, Delete) are recorded in the Log/Changes tables
- [x] Any database schema changes are implemented via `alembic revision` [transitions](https://pepys-import.readthedocs.io/en/latest/database_migration.html#how-to-use-it-for-developers)
- [x] I have completed the mandatory sections of this document.
- [x] I have deleted any unused sections.

### CODE REVIEW

**Comment by IanMayo** [comment]
Thanks for this Robin.

This isn't an intended UI-flow, but for a serial when one force is empty, it throws an error if `edit`, `delete` or `switch force` are clicked.  One way of handling it is to check there's an item selectd before trying to do the processing, another is to only enable these buttons if the list is non-empty and an item is selected.

Beyond that, this error got thrown when trying to delete a selected serial participant:
image

**Comment by robintw** [comment]
I'm pretty sure the bug you sent a screenshot of was fixed in #861. Could you try it there?

The other bug sounds very similar to the bug reported in https://github.com/debrief/pepys-import/pull/861#issuecomment-827531845 - I'll fix them both together, in that PR.

**Comment by IanMayo** [comment]
> I'm pretty sure the bug you sent a screenshot of was fixed in #861. Could you try it there?
> 
> The other bug sounds very similar to the bug reported in [#861 (comment)](https://github.com/debrief/pepys-import/pull/861#issuecomment-827531845) - I'll fix them both together, in that PR.

Yes - they did seem familiar.  If they're ready for review/merge I'm happy to do that, but they're currently both in `Draft`.

**Comment by IanMayo** [comment]
Thanks @robintw - will re-review once deployment available.

**Comment by IanMayo** [comment]
@robintw - I'm re-downloading the newest version, just in case I have an old one by mistake.  But in the release in this folder we still get an error if clicking on `Switch force` with no participant selected:
`pepys_import-20210428-switch-forces-c0798ec`

It would be understandable for it to be missed when fixing `disable buttons if no selection` in the other branch, since `<Switch force>` wasn't in that branch.

**Comment by robintw** [comment]
Sorry that was my fault, rushing before going to a meeting. I'd implemented the check for the edit and delete buttons, but not the switch force button. Fixed now - in this branch.

(I've got to go and deal with some urgent family stuff, but will check later today to see if anything else needs doing. If everything seems to work ok then this branch and #861 can be merged).

**Review by IanMayo — APPROVED** [review]
All great, thanks.

**Review by IanMayo — APPROVED** [review]
Working great, thanks.

File: pepys_admin/maintenance/tasks_gui.py [hunk 1]
```diff
@@ -257,6 +258,45 @@ def validate_fields(self, current_task, updated_fields):
...
 
+    def handle_duplicate(self):
+        current_task = self.task_edit_widget.task_object
+
+        # Work out a new name, so that if we copy multiple times we will get XXX Copy, XXX Copy 2 etc
+        all_serial_numbers_of_this_wargame = [
+            el.text for el in self.tree_view.selected_element.parent.children
+        ]
+
+        new_name_orig = current_task.serial_number + " Copy"
```
Comment (IanMayo) [inline pepys_admin/maintenance/tasks_gui.py:269]: this is clever :-)

File: pepys_admin/maintenance/widgets/participants_widget.py [hunk 2]
```diff
@@ -417,12 +430,57 @@ def handle_delete_button(self):
...
+        else:
+            new_force_type = ds.search_force_type("Blue")
+
+        participant.force_type = new_force_type
+
+        with ds.session_scope():
+            participant = ds.session.merge(participant)
+
+            change_id = ds.add_to_changes(
+                USER, datetime.utcnow(), "Manual delete from Tasks GUI"
```
Comment (IanMayo) [inline pepys_admin/maintenance/widgets/participants_widget.py:460]: I think this comment needs correcting to reflect the force change

File: pepys_admin/maintenance/tasks_gui.py [hunk 3]
```diff
@@ -257,6 +258,45 @@ def validate_fields(self, current_task, updated_fields):
...
+        new_serial = clone_model(current_task, serial_number=new_name)
+
+        with self.data_store.session_scope():
+            self.data_store.session.add(new_serial)
+            # Commit here, so that the new serial gets an ID, which we can reference below
+            self.data_store.session.commit()
+            self.data_store.session.refresh(new_serial)
+
+            new_serial_id = new_serial.serial_id
+            logger.debug(f"{new_serial_id=}")
```
Comment (IanMayo) [inline pepys_admin/maintenance/tasks_gui.py:285]: We can probably delete this logging statement.

**Comment by IanMayo** [comment]
@robintw  - have added some cosmetic comments, for fixing when you get chance.  I'm still going to use this branch to produce a new release - so they have these new features.

**Pull request merged**